android/app/src/main/java/io/locha/p2p/runtime/RuntimeEvents.java

```diff
@@ -136,4 +136,11 @@
      * @param peer The peer that is being dialed.
      */
     void onDialing(String peer);
+
+    /**
+     * external Address event
+     *
+     * @param  address
+     */
+    void onExternalAddress(String address);
```

Comment:
This is not necessary, it isn't called from Rust (which is what the interface is for), this can be a normal function (without `@Override`) inside `EventsDispatcher` class.